crates/net/p2p/src/error.rs

```diff
@@ -126,14 +126,11 @@ impl From<oneshot::error::RecvError> for RequestError {
     }
 }
 
-#[cfg(feature = "std")]
-impl std::error::Error for RequestError {}
-
 /// The download result type
 pub type DownloadResult<T> = Result<T, DownloadError>;
 
 /// The downloader error type
-#[derive(Debug, Clone, PartialEq, Eq, Display)]
+#[derive(Debug, Clone, PartialEq, Eq, Display, Error)]
```

Comment:
If I read the docs correctly, then this requires source attributes `#[error(source)]`

Reply:
Oh yeah just fixed and also I fixed the one related to `ConsensusError` after https://github.com/paradigmxyz/reth/pull/10831 because I think we missed the same thing here.